site/zenodo_rdm/migrator/load.py

```diff
@@ -0,0 +1,93 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2022-2023 CERN.
+#
+# ZenodoRDM is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Zenodo migrator files loader."""
+
+import contextlib
+from pathlib import Path
+
+import psycopg
+from invenio_rdm_migrator.load.postgresql import _ts
+from invenio_rdm_migrator.streams.files.models import (
+    FilesBucket,
+    FilesInstance,
+    FilesObjectVersion,
+)
+
+
+class ZenodoFilesLoad:
+    """Zenodo to RDM Files class for data loading."""
+
+    def __init__(self, db_uri, tmp_dir, **kwargs):
+        """Constructor."""
+        self.db_uri = db_uri
+        self.tmp_dir = Path(tmp_dir)
+        self.models = [FilesInstance, FilesBucket, FilesObjectVersion]
+
+    def _load(self):
+        """Bulk load CSV table files.
+
+        Loads the tables in the order given by the generator.
+        """
+
+        def get_cols(model):
+            from dataclasses import fields
+
+            return ", ".join([f.name for f in fields(model)])
+
+        with psycopg.connect(self.db_uri) as conn:
+            for model in self.models:
```

Comment:
shelve: from a quick look this is the only part that seems to be different from the original `PostgreSQLCopyLoad`. Could we abstract this somehow to avoid the duplication? Or move the inner loop into a separate underscore-method?